### PR TITLE
feat: update team members

### LINKED ISSUE #619: [FEATURE]: update team members
## Summary

As a team admin or super admin, I can add or remove team members. For this to be possible, the button "Add member" will be available, but only if the user is a team admin or super admin. When this button is clicked, a dialog will open with all the users: if the users belongs to the team, their checkbox is checked. When the user finished to add/remove members, when he clicks in the save/add button, a request is sent to the backend, to update the team users

### PR DESCRIPTION
**Pull request by patricia-mdias**
Relates to #619 

## Screenshots (if visual changes)
F17EB745-CB49-4904-9E38-F4FD7A09C42F
2DA8273F-C62E-4F62-B8DB-FBC5FED13234
3A32055A-591B-4F0B-A3DE-77327B9CFF75

## Proposed Changes

  - Team admin, super admins and stakeholders can add and/or remove team members.
  - Team admin and stakeholder can't edit their own new joiner status.
  - Change wording on adding/removing team members. The component used to remove and add members, but only said add team members - the success toast: team members added (when they were removed and/or added).

@CatiaAntunes96 

This pull request closes #619

### CODE REVIEW

File: frontend/src/components/Teams/CreateTeam/CardMember/index.tsx [hunk 1]
```diff
@@ -76,14 +77,26 @@ const CardMember = React.memo<CardBodyProps>(
...
+              ) : (
+                <Icon
+                  name="blob-personal"
+                  css={{
+                    width: '32px',
+                    height: '$32',
+                    zIndex: 1,
+                  }}
+                />
+              )}
```
Comment (nunocaseiro) [inline frontend/src/components/Teams/CreateTeam/CardMember/index.tsx:99]:
```suggestion
                <Icon
                  name="blob-personal"
                  css={{
                    width: '32px',
                    height: '$32',
                    zIndex: 1,
                    opacity: isOpen ? 0.2 : 1,
                  }}
                />
```

I'm not sure if the opacity is 1 when the `isOpen` is false but I know you got the idea

File: frontend/src/pages/teams/[teamId].tsx [hunk 2]
```diff
@@ -16,28 +16,68 @@ import TeamHeader from '@/components/Teams/Team/Header';
...
+    }));
+
+    const allUsersList: UserList[] = usersData.map((user: User) => ({
+      _id: user._id,
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      isSAdmin: user.isSAdmin,
+      joinedAt: user.joinedAt,
+      isChecked: true,
```
Comment (nunocaseiro) [inline frontend/src/pages/teams/[teamId].tsx:68]:
```suggestion
     ...user,
      isChecked: true,
```

File: frontend/src/hooks/useTeam.tsx [hunk 3]
```diff
@@ -108,6 +111,40 @@ const useTeam = ({ autoFetchTeam = false }: AutoFetchProps): UseTeamType => {
...
+
+      // updates members from the membersList recoil
+      setMembersList(finalMembersList);
+
+      setToastState({
+        open: true,
+        content: 'The team was successfully updated.',
+        type: ToastStateEnum.SUCCESS,
+      });
+    },
```
[the remark this answers is not in the document]
Comment (patricia-mdias) [reply frontend/src/hooks/useTeam.tsx:138]: I need the response from the endpoint to add the created users (with a teamUser id) to the membersList, how could I use `onMutate`and `onSettled` on this condition?

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀

**Pull request merged**